### PR TITLE
Initial attempt at function5 tests before implementation

### PR DESCRIPTION
**Pull request by jraza19**

### CODE REVIEW

File: tests/test_pygtracker.py [hunk 1]
```diff
@@ -22,6 +33,34 @@ def test_version():
 
 # Start tests for rank_students
 
+# tests that the user inputs the right number of students
+def test_number_of_students():
```
Comment (tdkhanhvu) [inline tests/test_pygtracker.py]: function name should be

test_functiontotest_conditionthatyoutest
```suggestion
def test_ranks_students_num_of_students_match():
```

File: tests/test_pygtracker.py [hunk 2]
```diff
@@ -22,6 +33,34 @@ def test_version():
...
+    assert (
+        gradetracker.rank_students(n=4).shape[0]
+    ) == 4, "Number of students don't match"
+
+
+# raises an error when a course id is not a part of the dataset
+def test_course_input():
```
Comment (tdkhanhvu) [inline tests/test_pygtracker.py]:
```suggestion
def test_rank_students_course_input_not_exist():
```

File: tests/test_pygtracker.py [hunk 3]
```diff
@@ -22,6 +33,34 @@ def test_version():
...
+    with raises(NameError):
+        gradetracker.rank_students(course_id="500", n=4)
+
+
+# dataframe is equal to an example dataframe
+def test_df_equal():
```
Comment (tdkhanhvu) [inline tests/test_pygtracker.py]: the test needs to have the real output to compare against. It is okay to write just a few tests first (that you expect to run and give you the output that matches what you expected)

File: tests/test_pygtracker.py [hunk 4]
```diff
@@ -22,6 +33,34 @@ def test_version():
...
+    example_df = pd.DataFrame(
+        {"student_id": "nameofstudent", "rank": 1, "grade": "addgradehere"}, index=[1]
+    )
+    assert example_df.equals(gradetracker.rank_students(course_id="511", n=1))
+
+
+# check the columns of the dataframe that come of the function
+def test_df_columns():
```
Comment (tdkhanhvu) [inline tests/test_pygtracker.py]: It's better to run the rank_students() once, assign to a variable, and then call that variable 3 times

student_ranks = gradetracker.rank_students(course_id="511", n=1)
assert rank.columns[0] == ....

Comment (tdkhanhvu) [inline tests/test_pygtracker.py]:
```suggestion
def test_rank_students_columns_names_match():
```

**Review by tdkhanhvu — CHANGES_REQUESTED** [review]

**Comment by jraza19** [comment]
Closing this issue as there has been a new PR (#39) made with the changes to reflect this.

**Pull request closed without merging**